runtime/src/sdk.rs

```diff
@@ -29,13 +29,43 @@ pub unsafe extern "C" fn extism_plugin_register(
         Err(e) => e.into_inner(),
     };
 
-    // Acquire lock and add plugin to registry
     plugins.push(plugin);
     let id = (plugins.len() - 1) as PluginIndex;
     info!("New plugin added: {id}");
     id
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn extism_plugin_update(
+    index: PluginIndex,
+    wasm: *const u8,
+    wasm_size: Size,
+    with_wasi: bool,
+) -> bool {
+    let index = index as usize;
+    trace!("Call to extism_plugin_replace with wasm pointer {:?}", wasm);
```

Comment:
```suggestion
    trace!("Call to extism_plugin_update with wasm pointer {:?}", wasm);
```

Reply:
although, "replace" is maybe more correct? since `Plugin::new` pretty much wipes away everything at the original plugin descriptor.

Reply:
Yeah I was on the fence about the naming, I still think update is a little more descriptive in the context of updating an existing plugin with a new version, but replace would also work.

Reply:
yea I like them both, no issue either way. update also opens up more things like update_manifest - maybe eventually other fields on the `Plugin` that make sense to be updated.